src/cryptoadvance/specter/helpers.py

```diff
@@ -307,3 +204,72 @@ def save_hwi_bridge_config(specter, config):
         config['whitelisted_domains'] = whitelisted_domains
     with open(os.path.join(specter.data_folder, 'hwi_bridge_config.json'), "w") as f:
         f.write(json.dumps(config, indent=4))
+
+def der_to_bytes(derivation):
+    items = derivation.split("/")
+    if len(items) == 0:
+        return b''
+    if items[0] == 'm':
+        items = items[1:]
+    if items[-1] == '':
+        items = items[:-1]
+    res = b''
+    for item in items:
+        index = 0
+        if item[-1] == 'h' or item[-1] == "'":
+            index += 0x80000000
+            item = item[:-1]
+        index += int(item)
+        res += index.to_bytes(4,'big')
+    return res
+
+def get_devices_with_keys_by_type(app, cosigners, wallet_type):
+    devices = []
+    prefix = "tpub"
+    if app.specter.chain == "main":
+        prefix = "xpub"
+    for cosigner in cosigners:
+        device = copy.deepcopy(cosigner)
+        allowed_types = ['', wallet_type]
+        device.keys = [key for key in device.keys if key.xpub.startswith(prefix) and key.key_type in allowed_types]
+        devices.append(device)
+    return devices
+
+def sort_descriptor(cli, descriptor, index=None, change=False):
```

Comment:
1. 8 spaces padding is wrong
2. `descriptor = descriptor.replace('sortedmulti', 'multi')` should be moved to the very beginning of the function, otherwise it will fail in `deriveaddress` if core v19 is used, and will get invalid ordering if v20 is used